core/types/block.go

```diff
@@ -307,7 +307,7 @@ func (b *Block) MixDigest() common.Hash   { return b.header.MixDigest }
 func (b *Block) Nonce() uint64            { return binary.BigEndian.Uint64(b.header.Nonce[:]) }
 func (b *Block) Bloom() Bloom             { return b.header.Bloom }
 func (b *Block) Coinbase() common.Address { return b.header.Coinbase }
-func (b *Block) Time() uint64             { return b.header.Time }
+func (b *Block) Time() *big.Int           { return b.header.Time }
```

Comment:
A block is meant to be immutable. Returning the internal time though makes it vulnerable as a big.Int is a pointer, not a value. Please make a copy and return that as the other big ints are returned from the block getters. (`return new(big.Int).Set(b.header.Time)`). See lines 300-303 (https://github.com/Gustav-Simonsson/go-ethereum/blob/timestamp_big_int/core/types/block.go#L300).